cpp/src/tsne/distances.h

```diff
@@ -78,29 +79,30 @@ void normalize_distances(const int n, float *distances, const int n_neighbors,
  * @input param indices: The input sorted indices from KNN.
  * @input param n: The number of rows in the data X.
  * @input param k: The number of nearest neighbors you want.
- * @input param P_sum: The sum of P.
  * @input param exaggeration: How much early pressure you want the clusters in TSNE to spread out more.
- * @output param COO_Matrix: The final P + P.T output COO matrix.
+ * @output param VAL: Values of COO Matrix
+ * @output param COL: Column indices of COO Matrix
+ * @output param ROW: Row indices of COO Matrix
+ * @input param row_sizes: Size (2*n) array for CSR style Row pointers. Used in symmetrization
  * @input param stream: The GPU stream.
  * @input param handle: The GPU handle.
  */
 template <int TPB_X = 32>
 void symmetrize_perplexity(float *P, long *indices, const int n, const int k,
-                           const float P_sum, const float exaggeration,
-                           MLCommon::Sparse::COO<float> *COO_Matrix,
+                           const float exaggeration,
+                           /* MLCommon::Sparse::COO<float> *COO_Matrix, */
```

Comment:
Remove commented code. If you don't mind my asking- why drop the COO container format? It simplifies the function headers immensely. I've got an open PR to use device_buffer underneath COO & CSR so that should make them cleaner to use as well.

Reply:
Ohhh is it possible to construct a COO_Matrix from VAL, ROW, COL pointers? Can I just do COO_Matrix->rows = ROW etc?

Reply:
Currently there are constructors for this. I have a PR that updates the COO to use device_buffers, so those could be passed in directly.

Reply:
Oh cool :)